src/profile-logic/profile-data.js

```diff
@@ -797,6 +798,99 @@ export function getSearchFilteredMarkers(
   return newMarkersTable;
 }
 
+/**
+ * This function takes a marker that packs in a marker payload into the string of the
+ * name. This extracts that and turns it into a payload.
+ */
+export function extractMarkerDataFromName(thread: Thread): Thread {
+  const { stringTable, markers } = thread;
+  const newMarkers: MarkersTable = {
+    data: markers.data.slice(),
+    name: markers.name.slice(),
+    time: markers.time.slice(),
+    length: markers.length,
+  };
+
+  // Match: "Bailout_MonitorTypes after add on line 1013 of self-hosted:1008"
+  // Match: "Bailout_TypeBarrierO at jumptarget on line 1490 of resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:1484"
+  const bailoutRegex =
+    // Capture groups:
+    //       type   afterAt    where   bailoutLine  script functionLine
+    //        ↓     ↓          ↓             ↓        ↓    ↓
+    /^Bailout_(\w+) (after|at) ([\w ]+) on line (\d+) of (.*):(\d)+$/;
+
+  // Match: "Invalidate resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:3662"
+  // Match: "Invalidate self-hosted:4032"
+  const invalidateRegex =
+    // Capture groups:
+    //         url    line
+    //           ↓    ↓
+    /^Invalidate (.*):(\d)+$/;
+
+  const bailoutStringIndex = stringTable.indexForString('Bailout');
+  const invalidationStringIndex = stringTable.indexForString('Invalidate');
+  for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+    const nameIndex = markers.name[markerIndex];
+    const time = markers.time[markerIndex];
+    const name = stringTable.getString(nameIndex);
+    let matchFound = false;
+    if (name.startsWith('Bailout_')) {
```

Comment:
As a nit, we could try to match against the regexp already here instead of using `startsWith`. But I found the error below nice to work with, + this is likely faster to first use `startsWith` in case this doesn't match. So let's keep it like this.